plugins/inputs/openweathermap/openweathermap.go

```diff
@@ -0,0 +1,276 @@
+	Dt     int64  `json:"dt"`
+	Dttxt  string `json:"dt_txt"` // empty for weather/
+	Clouds struct {
+		All int64 `json:"all"`
+	} `json:"clouds"`
+	Main struct {
+		GrndLevel float64 `json:"grnd_level"` // empty for weather/
+		Humidity  int64   `json:"humidity"`
+		SeaLevel  float64 `json:"sea_level"` // empty for weather/
+		Pressure  float64 `json:"pressure"`
+		Temp      float64 `json:"temp"`
+		TempMax   float64 `json:"temp_max"`
+		TempMin   float64 `json:"temp_min"`
+	} `json:"main"`
+	Rain struct {
+		Rain3 float64 `json:"3h"`
+	} `json:"rain"`
+	Sys struct {
+		Pod     string  `json:"pod"`
+		Country string  `json:"country"`
+		Message float64 `json:"message"`
+		Id      int64   `json:"id"`
+		Type    int64   `json:"type"`
+		Sunrise int64   `json:"sunrise"`
+		Sunset  int64   `json:"sunset"`
+	} `json:"sys"`
+	Wind struct {
+		Deg   float64 `json:"deg"`
+		Speed float64 `json:"speed"`
+	} `json:"wind"`
+	Weather []struct {
+		Description string `json:"description"`
+		Icon        string `json:"icon"`
+		Id          int64  `json:"id"`
+		Main        string `json:"main"`
+	} `json:"weather"`
+
+	// Additional entries for weather/
+	Id    int64  `json:"id"`
+	Name  string `json:"name"`
+	Coord struct {
+		Lat float64 `json:"lat"`
+		Lon float64 `json:"lon"`
+	} `json:"coord"`
+	Visibility int64 `json:"visibility"`
+}
+
+type Status struct {
+	City struct {
+		Coord struct {
+			Lat float64 `json:"lat"`
+			Lon float64 `json:"lon"`
+		} `json:"coord"`
+		Country string `json:"country"`
+		Id      int64  `json:"id"`
+		Name    string `json:"name"`
+	} `json:"city"`
+	List []WeatherEntry `json:"list"`
+}
+
+func gatherStatusUrl(r *bufio.Reader, tags map[string]string, acc telegraf.Accumulator) error {
+	dec := json.NewDecoder(r)
+	status := &Status{}
+	if err := dec.Decode(status); err != nil {
+		return fmt.Errorf("Error while decoding JSON response: %s", err)
+	}
+	status.Gather(tags, acc)
+	return nil
+}
+
+func gatherWeatherUrl(r *bufio.Reader, tags map[string]string, acc telegraf.Accumulator) error {
+	dec := json.NewDecoder(r)
+	status := &Status{}
+	e := WeatherEntry{}
+	if err := dec.Decode(&e); err != nil {
+		return fmt.Errorf("Error while decoding JSON response: %s", err)
+	}
+	status.List = make([]WeatherEntry, 0)
+	if len(e.Name) > 0 {
+		status.List = append(status.List, e)
+		status.City.Coord.Lat = e.Coord.Lat
+		status.City.Coord.Lon = e.Coord.Lon
+		status.City.Id = e.Id
+		status.City.Name = e.Name
+		status.Gather(tags, acc)
+	}
+	return nil
+}
+
+func (s *Status) Gather(tags map[string]string, acc telegraf.Accumulator) {
+	tags["city_id"] = strconv.FormatInt(s.City.Id, 10)
+	for _, e := range s.List {
+		tm := time.Unix(e.Dt, 0)
+		acc.AddFields(
+			"weather",
+			map[string]interface{}{
+				"rain":        e.Rain.Rain3,
+				"wind.deg":    e.Wind.Deg,
+				"wind.speed":  e.Wind.Speed,
+				"humidity":    e.Main.Humidity,
+				"pressure":    e.Main.Pressure,
+				"temperature": e.Main.Temp - 273.15, // Kelvin to Celsius
```

Comment:
adding units format would remove the need for this conversion

Reply:
See previous question

Reply:
Also, remove this conversion.